src/app/services/streetview-authentication.service.ts

```diff
@@ -0,0 +1,242 @@
+import { EnvService } from '../services/env.service';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { NotificationsService } from './notifications.service';
+import { tap } from 'rxjs/operators';
+import { StreetviewRequest, Streetview } from '../models/streetview';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class StreetviewAuthenticationService {
+  private _activeStreetview: BehaviorSubject<Streetview> = new BehaviorSubject(
+    null
+  );
+  public activeStreetview$: Observable<Streetview> =
+    this._activeStreetview.asObservable();
+  private _streetviews: BehaviorSubject<Array<Streetview>> =
+    new BehaviorSubject([]);
+  public streetviews$: Observable<Array<Streetview>> =
+    this._streetviews.asObservable();
+
+  private TOKEN_STORE = {
+    google: 'googleToken',
+    mapillary: 'mapillaryToken',
+  };
+  private _username: string;
+
+  constructor(
+    private http: HttpClient,
+    private authService: AuthService,
+    private notificationsService: NotificationsService,
+    private envService: EnvService,
+    private router: Router
+  ) {
+    this.authService.currentUser.subscribe(
+      (u) => (this._username = u ? u.username : null)
+    );
+  }
+
+  public isLoggedIn(service: string): boolean {
+    const userToken: AuthToken = this.getLocalToken(service);
+
+    if (!this._activeStreetview.getValue()) {
+      return false;
+    }
+
+    return userToken && !userToken.isExpired();
+  }
+
+  public login(service: string, projectId: number) {
+    this.tokenRequest(service, projectId);
+  }
+
+  public logout(service: string): void {
+    localStorage.removeItem(this.TOKEN_STORE[service]);
+    this.deleteStreetviewByService(service);
+  }
+
+  public sequenceInStreetview(sequenceId: string): boolean {
+    return this._activeStreetview.getValue().instances.some((instance) =>
+      instance.sequences.some((sequence) => {
+        return sequence.sequence_id === sequenceId;
+      })
+    );
+  }
+
+  public getStreetviews(): Observable<Array<Streetview>> {
+    return this.http
+      .get<Array<Streetview>>(this.envService.apiUrl + `/streetview/`)
+      .pipe(
+        tap((streetviews: Array<Streetview>) => {
+          this._streetviews.next(streetviews);
+          // NOTE: With just mapillary, assume single active streetview
+          this._activeStreetview.next(streetviews[0]);
+          console.log(this._activeStreetview);
+        })
+      );
+  }
+
+  public createStreetview(data: StreetviewRequest) {
+    return this.http
+      .post<any>(this.envService.apiUrl + '/streetview/', data)
+      .pipe(
+        tap(() => {
+          this.getStreetviews().subscribe();
+        })
+      );
+  }
+
+  public updateStreetview(streetviewId: number, data: StreetviewRequest) {
+    return this.http
+      .put<any>(this.envService.apiUrl + `/streetview/${streetviewId}/`, data)
+      .subscribe(() => {
+        this.getStreetviews().subscribe();
+      });
+  }
```

Comment:
```suggestion
```

unused; should be dropped.   looks like 'updateStreetviewByService` is what is used and supported in backend.

Reply:
Ignore above comment. lets discuss elsewhere